Svc/FpySequencer/FpySequencer.cpp

```diff
@@ -0,0 +1,272 @@
+// ======================================================================
+// \title  FpySequencer.cpp
+// \author zimri.leisher
+// \brief  cpp file for FpySequencer component implementation class
+// ======================================================================
+
+#include <Svc/FpySequencer/FpySequencer.hpp>
+
+namespace Svc {
+
+// ----------------------------------------------------------------------
+// Construction, initialization, and destruction
+// ----------------------------------------------------------------------
+
+FpySequencer ::FpySequencer(const char* const compName) : 
+    FpySequencerComponentBase(compName), 
+    m_sequenceBuffer(),
+    m_allocatorId(0),
+    m_sequenceFilePath("<invalid_seq>"),
+    m_sequenceObj(),
+    m_computedCRC(0),
+    m_sequenceBlockState(),
+    m_savedOpCode(0),
+    m_savedCmdSeq(0),
+    m_goalState(),
+    m_sequencesStarted(0),
+    m_statementsDispatched(0),
+    m_runtime(),
+    m_tlm()
+{}
+
+FpySequencer ::~FpySequencer() {}
+
+//! Handler for command RUN
+//!
+//! Loads, validates and runs a sequence
+void FpySequencer::RUN_cmdHandler(FwOpcodeType opCode,               //!< The opcode
+                                  U32 cmdSeq,                        //!< The command sequence number
+                                  const Fw::CmdStringArg& fileName,  //!< The name of the sequence file
+                                  FpySequencer_BlockState block      //!< Return command status when complete or not
+) {
+    // can only run a seq while in idle
+    if (sequencer_getState() != State::IDLE) {
+        this->log_WARNING_HI_InvalidCommand(static_cast<I32>(sequencer_getState()));
+        this->cmdResponse_out(opCode, cmdSeq, Fw::CmdResponse::EXECUTION_ERROR);
+        return;
+    }
+
+    if (block == FpySequencer_BlockState::BLOCK) {
+        // save the opCode and cmdSeq so we can respond later
+        this->m_savedOpCode = opCode;
+        this->m_savedCmdSeq = cmdSeq;
+    }
+
+    this->sequencer_sendSignal_cmd_RUN(FpySequencer_SequenceExecutionArgs(fileName, block));
+
+    // only respond if the user doesn't want us to block further execution
+    if (block == FpySequencer_BlockState::NO_BLOCK) {
+        this->cmdResponse_out(opCode, cmdSeq, Fw::CmdResponse::OK);
+    }
+}
+
+//! Handler for command VALIDATE
+//!
+//! Loads and validates a sequence
+void FpySequencer::VALIDATE_cmdHandler(FwOpcodeType opCode,              //!< The opcode
+                                       U32 cmdSeq,                       //!< The command sequence number
+                                       const Fw::CmdStringArg& fileName  //!< The name of the sequence file
+) {
+    // can only validate a seq while in idle
+    if (sequencer_getState() != State::IDLE) {
```

Comment:
While logically this makes sense, is there any technical reason one couldn't pre-validate a sequence while another is running?

Reply:
Curiosity, not blocking!